linera-execution/src/runtime.rs

```diff
@@ -1030,51 +1042,82 @@ impl<UserInstance> Clone for SyncRuntimeHandle<UserInstance> {
 }
 
 impl ContractSyncRuntime {
-    /// Main entry point to start executing a user action.
-    #[expect(clippy::too_many_arguments)]
-    pub(crate) fn run_action(
+    pub(crate) fn new(
         execution_state_sender: ExecutionStateSender,
-        application_id: UserApplicationId,
         chain_id: ChainId,
         local_time: Timestamp,
         refund_grant_to: Option<Account>,
         resource_controller: ResourceController,
-        action: UserAction,
+        action: &UserAction,
```

Comment:
This seems risky. It would be possible to create an instance using one action and then run a different action. I'd prefer to move the action into the type here in the constructor.

Reply:
It feels a bit weird to have a runtime that only supports one action. Do you think it makes sense to move the action out of the internal runtime?

Reply:
I think there were discussions about this before. Right now the runtime has a memory state that only lives while a transaction (the `UserAction`) is executed, so it makes sense to only have one. We considered making it have a "block-long" lifetime, and that would be better if the runtime supported multiple actions during its existence. However, I'm not sure we'll go forward with that (even the long-lived service is feature gated for now), so I think it wouldn't make sense to try to make it support more than one action now :thinking:

Reply:
Unfortunately because the `SyncRuntime` (of which the `ContractSyncRuntime` is a specialization) is generic over its bytecode type, and service runtimes don't have built-in actions, it's rather tricky to add to the type :thinking:

Reply:
We agreed to deal with this for now by just adding some asserts as guard rails.